preview-docs/GraphRoles.md

```diff
@@ -9,10 +9,10 @@ We've identified new requirements for access control at the graph level since th
-2. Introduce a way to override any role a user might have in the org with a higher role for a specific graph. A user might be an **Observer** in the org, but can be promoted to a **Contributor** on a specific graph. When users create a graph, they are automatically made **Graph Admin** on that graph.
-3. Introduce a way to mark variants as **protected**. On a protected variant, you need to be a **Graph Admin** rather than a **Contributor** to push new versions of the graph's schema to that variant.
-4. Introduce a way to assign roles to graph API keys. We would make the **Consumer**, **Observer**, **Contributor**, and **Graph Admin** roles available for assignment on API key creation. (Existing graph API keys would act as **Contributor**s.)
-5. Introduce a setting on graphs that flags them as "visible" or "hidden" to the members of that graph's org by default. Right now, all graphs in the org are visible to all members of the org. If you mark a graph as "hidden" to your org, only users with explicit overridden roles on the graph (as described in "2") or who are **Org Admin**s can see it. (While **Org Admin**s will have full read and write access to hidden graphs, hidden graphs that they do not have explicit overridden roles on may still be hidden by default from graph lists in the UI unless explicitly requested.)
+2. Introduce a way to override any role a user might have in the org with a higher role for a specific graph. A user might be an **Observer** in the org, but can be promoted to a **Contributor** on a specific graph. When users create a graph, they are automatically made **Graph Admin** on that graph. **Graph Admin**s and **Org Admin**s can update these graph-specific permissions.
+3. Introduce a way to mark variants as **protected**. On a protected variant, you need to be a **Graph Admin** rather than a **Contributor** to push new versions of the graph's schema to that variant. **Graph Admin**s and **Org Admin**s can control whether variants are protected.
+4. Introduce a way to assign roles to graph API keys. We would make the **Consumer**, **Observer**, **Contributor**, and **Graph Admin** roles available for assignment on API key creation. (Existing graph API keys would act as **Contributor**s.) Users can create and view keys with roles that have power equal to or less than their role on the graph.
```

Comment:
Alternatively, "Only **Graph Admin**s and **Org Admin**s can create and view graph API keys."

That would probably be simpler since either you can view all graph API keys or no graph API keys. Thoughts?

Reply:
Hmm, so you don't think it'd be useful for Contributors to make graph scooped keys? I suppose I can't think of a use case where user keys wouldn't work...

Reply:
I think it might be a useful thing to support. It's also a bit of a challenge in our current role system where all the roles are "can I do a class of activities to a graph", not "can I do a specific activity with parameter X to a graph". eg would we have a ServiceRoles GraphQL field `ServiceRoles.canQueryTokens(forRole: UserPermission)`? Or would the filtering just be done outside of the roles system? Hopefully that lets us build the right UI.